lib/token.js

```diff
@@ -137,7 +138,10 @@ class Token {
     }
 
     if (result) {
-      const val = result[0]
+      let val = result
+      if (BigNumber.isBigNumber(result)) {
```

Comment:
When transitioning from @metamask/ethjs-contract to @ethersproject/contracts, the results are returned in the @ethersproject/bignumber format. Here, we convert them back to BN to prevent any downstream incompatibilities.

Reply:
Makes sense, good call!